From 20c818a3f1cda0124c6a1d942c5e7468c78c8ce5 Mon Sep 17 00:00:00 2001
From: Gabor Igloi <gabor.igloi@citrix.com>
Date: Fri, 20 Oct 2017 22:55:01 +0100
Subject: [PATCH] switch/clock: use only Mtime_clock to satisfy CLOCK signature
 of shared-block-ring

Remove the fallback to Unix functions - Mtime_clock supports platformst
with a POSIX clock, including Linux.

Signed-off-by: Gabor Igloi <gabor.igloi@citrix.com>
---
 switch/clock.ml  | 18 +-----------------
 switch/clock.mli |  8 +-------
 2 files changed, 2 insertions(+), 24 deletions(-)

diff --git a/switch/clock.ml b/switch/clock.ml
index 04eae1a..8ff49d4 100644
--- a/switch/clock.ml
+++ b/switch/clock.ml
@@ -14,20 +14,4 @@
  * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  *)
 
-let use_mtime () = Mtime.Span.to_uint64_ns (Mtime_clock.elapsed ())
-
-let use_timeofday () = Int64.of_float (Unix.gettimeofday () *. Mtime.s_to_ns)
-
-let ns =
-  try
-    Mtime_clock.now () |> ignore;
-    use_mtime
-  with Sys_error e -> begin
-      Logging.warn "Error: %s. No monotonic clock source: falling back to calendar time" e;
-      use_timeofday
-    end
-
-include Unix
-let time = gettimeofday
-
-let s () = Int64.to_float (ns ()) /. 1e9
+include Mtime_clock
diff --git a/switch/clock.mli b/switch/clock.mli
index 70f55de..7f2f306 100644
--- a/switch/clock.mli
+++ b/switch/clock.mli
@@ -14,10 +14,4 @@
  * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  *)
 
-include V1.CLOCK
-
-val ns: unit -> int64
-(** [ns ()] time in ns since the program started *)
-
-val s: unit -> float
-(** [s ()] time in seconds since the program started *)
+include Shared_block.S.CLOCK
-- 
2.11.0

